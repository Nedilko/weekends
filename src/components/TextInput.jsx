import PropTypes from 'prop-types'

TextInput.propTypes = {
  placeholder: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
}

function TextInput({ placeholder, value, onChange }) {
  return (
    <input
      type="text"
      placeholder={placeholder}
      maxLength="20"
      value={value}
      onChange={(e) => {
        onChange(e.target.value)
      }}
      className="flex focus:ring-slate-500 focus:border-slate-500 border-gray-300 border rounded-md mx-2 px-2 font-thin text-xl text-slate-600 dark:text-gray-400"
    />
  )
}

export default TextInput
