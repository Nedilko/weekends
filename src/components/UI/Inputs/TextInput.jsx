import PropTypes from 'prop-types'

TextInput.propTypes = {
  placeholder: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
}

function TextInput({ placeholder, value, onChange }) {
  return (
    <input
      type="text"
      placeholder={placeholder}
      maxLength="20"
      value={value}
      onChange={(e) => {
        onChange(e.target.value)
      }}
      className="mx-2 flex rounded-md border border-slate-100 px-2 text-xl font-thin text-slate-600 focus:border-slate-300 focus:outline-none dark:border-gray-700 dark:bg-zinc-900 dark:text-gray-400 dark:focus:border-gray-500"
    />
  )
}

export default TextInput
