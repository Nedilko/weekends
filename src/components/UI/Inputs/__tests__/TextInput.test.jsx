import TextInput from '@UI/Inputs/TextInput'
import { render, screen, userEvent } from '@utils/test-utils'

describe('TextInput', () => {
  const changeHandler = vi.fn()

  beforeEach(() => {
    changeHandler.mockClear()
  })

  it('should render valid', () => {
    render(
      <TextInput
        placeholder="sample placeholder"
        value="sample"
        onChange={changeHandler}
      />
    )
    expect(screen.getByTestId('greetings-text-input')).toMatchSnapshot()
  })

  it('should render invalid', async () => {
    const user = userEvent.setup()
    render(
      <TextInput
        placeholder="sample placeholder"
        value="sample"
        onChange={changeHandler}
      />
    )
    const input = screen.getByTestId('greetings-text-input')
    await user.clear(input)
    expect(input).toMatchSnapshot()
    await user.type(input, '   ')
    expect(input).toMatchSnapshot()
  })

  it('should call change handler with typed value', async () => {
    const user = userEvent.setup()
    render(
      <TextInput
        placeholder="sample placeholder"
        value="sample"
        onChange={changeHandler}
      />
    )
    const input = screen.getByTestId('greetings-text-input')
    await user.clear(input)
    await user.type(input, 'have a beer')
    expect(changeHandler).toHaveBeenCalledWith('have a beer')
  })
})
