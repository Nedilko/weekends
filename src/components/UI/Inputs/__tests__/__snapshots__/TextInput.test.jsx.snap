// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render invalid 1`] = `
<input
  class="border-red-500 mx-2 flex rounded-md border px-2 text-xl font-thin text-slate-600 focus:outline-none dark:border-gray-700 dark:bg-zinc-900 dark:text-gray-400 dark:focus:border-gray-500"
  data-testid="greetings-text-input"
  maxlength="20"
  placeholder="sample placeholder"
  type="text"
  value=""
/>
`;

exports[`TextInput should render invalid 2`] = `
<input
  class="border-red-500 mx-2 flex rounded-md border px-2 text-xl font-thin text-slate-600 focus:outline-none dark:border-gray-700 dark:bg-zinc-900 dark:text-gray-400 dark:focus:border-gray-500"
  data-testid="greetings-text-input"
  maxlength="20"
  placeholder="sample placeholder"
  type="text"
  value="   "
/>
`;

exports[`TextInput should render valid 1`] = `
<input
  class="border-slate-100 focus:border-slate-300 mx-2 flex rounded-md border px-2 text-xl font-thin text-slate-600 focus:outline-none dark:border-gray-700 dark:bg-zinc-900 dark:text-gray-400 dark:focus:border-gray-500"
  data-testid="greetings-text-input"
  maxlength="20"
  placeholder="sample placeholder"
  type="text"
  value="sample"
/>
`;
